Header header
AgentPath[] path_array